.flake8

```diff
@@ -0,0 +1,2 @@
+[flake8]
```

Comment:
@tzaffi For https://www.flake8rules.com/rules/F405.html and https://www.flake8rules.com/rules/F403.html, I'm imagining we'd prefer to avoid the violation in source code.  Does it work for you to limit ignoring F403 + F405 to `*_test.py` and `examples/`?

From the previous attempt, there's an example ignoring `*_test.py`:  https://github.com/algorand/pyteal/pull/213/files#diff-6951dbb399883798a226c1fb496fdb4183b1ab48865e75eddecf6ceb6cf46442R11. 

Running locally, I see there's a F403 violation in source code.  I didn't look closely enough to determine how easily it can be avoided vs adding inline ignore.  I can take a deeper look if you'd like - let me know.

```
~/dev/pyteal make-and-flake *1 !1 ?6 ────────────────────────────────────────────────────────────────────────── ✘ 2|1 5s  pyteal 09:53:33 AM
❯ make flake8 | grep -v -E 'test|examples'
pyteal/__init__.py:1:1: F403 'from .ast import *' used; unable to detect undefined names
pyteal/__init__.py:3:1: F403 'from .ir import *' used; unable to detect undefined names
make: *** [flake8] Error 1
```

Reply:
I'll research this some more and update here.

Reply:
When I first joined I didn't like our profligate use of * imports in pyteal. But the more I use pyteal, the more this convention makes sense to me. We have so many constructs that are typically used in pyteal, that a standard pyteal program would have 10 or more objects being imported. 

An alternate approach in such situations (where a library with many constructs is used) is to abbreviate the imported module and then always refer to it. For example:

```python
import pandas as pd

...
df = pd.DataFrame(...)
```

we could follow a similar convention and start enforcing it here with usages such as below (notice, we are already implicitly recommending this approach with `abi`):

```python
import pyteal as pt

toSum = abi.DynamicArray(abi.Uint64TypeSpec())
toMultiply = abi.DynamicArray(abi.Uint64TypeSpec())
results = abi.Tuple(abi.Uint64TypeSpec(), abi.Uint64TypeSpec())
program = pt.Seq(
    toSum.decode(pt.Txn.application_args[0]),
    toMultiply.decode(pt.Txn.application_args[1]),
    sumAndMultiply(toSum, toMultiply).store_into(results)
)
```

Reply:
A few thoughts from my side:
* If folks are on-board, I am on-board to use the `pandas as pd` approach.  I think it'd help disambiguate PyTeal constructs that match Python standard library (e.g. `Array`).
* If folks prefer to keep as is, I'd advocate to add the ignore only for tests and examples.  I'm thinking in source code we can be more perspective.  Though I'll ultimately follow the group's recommendation here.

Reply:
There's 3 separate issues:
* Use of star imports in our runtime code: this is highly useful because we have a few modules that import and export _all_ of a submodule. It would be too time consuming and error prone to not use a star import here. My opinion is that adding inline exceptions for these cases is good enough.
* Use of star imports in example: from the beginning we've encouraged this because PyTeal programs can use a lot of different AST elements, and because our syntax is already clunky enough without added prefixed in my opinion. I'm in favor of keeping it as-is since our examples should look natural and not be intimidating.
* Use of star imports in our tests: this matters least to me, so if you'd like to adopt `import pyteal as pt` or similar, I wouldn't mind.

Reply:
I'll go with the modifying the tests files only and adding flake8 exceptions for other files.

Reply:
hehe... that resulted in 6000+ changes

Reply:
@tzaffi Thanks for the changes here - I think it's reasonable starting point and we can iterate over time.  Resolving.